### PR TITLE
feat(akamai): unzip user content

### PR DESCRIPTION
**Pull request by guilhem**
This pull request introduces functionality to handle compressed userdata in the Linode Metadata Service and includes the implementation of a utility function to decompress gzip data if needed. The most important changes include modifications to the `fetchConfig` function and the addition of a new utility function for gzip decompression.

Enhancements to userdata handling:

* [`internal/providers/akamai/akamai.go`](diffhunk://#diff-be714034d6f150da34c73bffef8c0abe1f40bb1120133acacc667da90b09678bL94-R101): Updated the `fetchConfig` function to decompress gzip data if needed before parsing the configuration.

New utility function:

* [`internal/providers/util/unzip.go`](diffhunk://#diff-6bfa7cb53910fd35effb0a56e78ac477cc39aec9b1d624c2992ae8a616ec0429R1-R43): Added a new utility function `GunzipIfNeeded` to handle gzip decompression, including a helper function `hasGzipMagicNumber` to check for gzip magic numbers.

### CODE REVIEW

File: internal/providers/util/unzip.go
Comment (eljohnson92) [inline internal/providers/util/unzip.go:32]: What do you think about inverting this logic so we can un-nest the bulk of the logic here ?
```
if !hasGzipMagicNumber(raw) { return raw, nil }
...
```

File: internal/providers/akamai/akamai.go
Comment (guilhem) [reply internal/providers/akamai/akamai.go:1]: done :)

**Comment by guilhem** [comment]
I will speak to the metadata team and ask about this header :)

**Comment by guilhem** [comment]
So the header will not be added soon.
Another problem with this header is that we are embedding gzip in base64, so content is more text than gzip.

What can we do to have this PR merged?
Can we see a work on fetcher for another PR?

**Review by eljohnson92 — APPROVED** [review]

**Comment by guilhem** [comment]
@jlebon can we have a review? :)

**Review by prestist — COMMENTED** [review]
Overall this looks great! Thank you for contributing and thank you for testing @tormath1 ❤️ 

Functionally it lgtm I have some small nits that I dont feel very strong about but would like to at least bring them up. 

And should we add a comment about https://github.com/coreos/ignition/pull/1934#issuecomment-2374409544 and link an rfe or somthing else

File: internal/providers/util/unzip.go
Comment (guilhem) [reply internal/providers/util/unzip.go]: I change the way to test unzip by using a standard function: [`http.DetectContentType`](https://pkg.go.dev/net/http#DetectContentType).

I also added test to be sure that it's working as expected

@prestist is it something better than with “magic” variables :)

Comment (prestist) [reply internal/providers/util/unzip.go]: @guilhem absolutely! 💯

**Comment by prestist** [comment]
@guilhem you have my approval, I just need to see green, looks like we have failing unit tests?

**Comment by guilhem** [comment]
> @guilhem you have my approval, I just need to see green, looks like we have failing unit tests?

Awesome 😎 
Sadly it's a failure in all PR.

**Comment by prestist** [comment]
@guilhem  if you rebase your pr should be good now

**Comment by guilhem** [comment]
@prestist rebase done.
I see Golang unit tests ok, so just waiting for all the other tests and it should be fine to merge.

**Pull request merged**
